raiden/network/transport/matrix.py

```diff
@@ -1062,21 +1093,61 @@ def _get_user(self, user: Union[User, str]) -> User:
         return user
 
     def _set_room_id_for_address(self, address: Address, room_id: Optional[str]):
-        """ Uses GMatrixClient.set_account_data to keep updated mapping of addresses->rooms """
+        """ Uses GMatrixClient.set_account_data to keep updated mapping of addresses->rooms
+
+        If room_id is falsy, clean list of rooms. Else, push room_id to front of the list """
+
+        assert not room_id or room_id in self._client.rooms, 'Invalid room_id'
         address_hex = to_checksum_address(address)
-        _address_to_room_id = self._client.account_data.get('network.raiden.rooms', {})
-        if room_id != _address_to_room_id.get(address_hex):
-            if room_id:
-                _address_to_room_id[address_hex] = room_id
-            else:
-                _address_to_room_id.pop(address_hex, None)
-            self._client.set_account_data('network.raiden.rooms', _address_to_room_id)
 
-    def _get_room_id_for_address(self, address: Address) -> Optional[str]:
-        """ Uses GMatrixClient.get_account_data to get updated mapping of addresses->rooms """
+        # no need to deepcopy, we don't modify lists in-place
+        # type: Dict[address_hex, List[room_id]]
```

Comment:
Why not explicitly specify the type and leave it in a comment?

Reply:
Because the type `Dict[str, List[str]]` is not as informative on what is in that dict as I'd like to

Reply:
Well you can achieve the same with types:

`Dict[typing.AddressHex, List[typing.RoomID]]

`RoomID` does not exist yet but it should. I had a lot of difficulty figuring out what type a room id is supposed to be when first looking at the code.

Reply:
I didn't add it before because these are implementation details of the matrix transport, but I'll do